spec/an_localize_libs_action_spec.rb

```diff
@@ -1,48 +1,223 @@
-    #{indented_content}
-    </resources>
-  XML
+  def write_android_xml(path, lines)
+    File.write(path, android_xml_with_lines(lines))
+  end
+
+  describe 'merges the strings from the given array into the given main strings file' do
+    it 'handles simple XMLs with no duplicates nor attributes' do
+      in_tmp_dir do |tmp_dir|
+        app_strings_path = File.join(tmp_dir, 'app.xml')
+        File.write(app_strings_path, android_xml_with_lines(['<string name="a_string">test from app</string>']))
+
+        lib1_strings_path = File.join(tmp_dir, 'lib1.xml')
+        File.write(lib1_strings_path, android_xml_with_lines(['<string name="a_lib1_string">test from lib 1</string>']))
+
+        lib2_strings_path = File.join(tmp_dir, 'lib2.xml')
+        File.write(lib2_strings_path, android_xml_with_lines(['<string name="a_lib2_string">test from lib 2</string>']))
+
+        run_described_fastlane_action(
+          app_strings_path: app_strings_path,
+          libs_strings_path: [
+            { library: 'lib_1', strings_path: lib1_strings_path, exclusions: [] },
+            { library: 'lib_2', strings_path: lib2_strings_path, exclusions: [] },
+          ]
+        )
+
+        expected = [
+          '<string name="a_string">test from app</string>',
+          '<string name="a_lib1_string">test from lib 1</string>',
+          '<string name="a_lib2_string">test from lib 2</string>',
+        ]
+        expect(File.read(app_strings_path)).to eq(android_xml_with_lines(expected))
+      end
+    end
+
+    it 'keeps app value if content_override is used' do
+      in_tmp_dir do |tmp_dir|
+        app_strings_path = File.join(tmp_dir, 'app.xml')
+        app_xml_lines = [
+          '<string name="override-true" content_override="true">from app override-true</string>',
+          '<string name="override-false" content_override="false">from app override-false</string>',
+          '<string name="override-missing">from app override-missing</string>',
+        ]
+        File.write(app_strings_path, android_xml_with_lines(app_xml_lines))
+
+        lib_strings_path = File.join(tmp_dir, 'lib.xml')
+        lib_xml_lines = [
+          '<string name="override-true">from lib override-true</string>',
+          '<string name="override-false">from lib override-false</string>',
+          '<string name="override-missing">from lib override-missing</string>',
+        ]
+        File.write(lib_strings_path, android_xml_with_lines(lib_xml_lines))
+
+        run_described_fastlane_action(
+          app_strings_path: app_strings_path,
+          libs_strings_path: [
+            { library: 'lib', strings_path: lib_strings_path, exclusions: [] },
+          ]
+        )
+
+        expected = [
+          '<string name="override-true" content_override="true">from app override-true</string>',
+          '', '', # FIXME: Current implementation adds empty lines; we should get rid of those at some point
```

Comment:
This will actually already be addressed by https://github.com/wordpress-mobile/release-toolkit/pull/358 (which builds on top of this PR) 🎉